tribe-ext-settings-import-export.php

```diff
@@ -0,0 +1,328 @@
+								<p>
+									<input type="text" name="import_reset_confirmation"/><br/>
+									<?php esc_html_e( 'Enter "reset" into the above field if you would like to reset the settings.', 'tribe-ext-settings-import-export' ); ?>
+								</p>
+								<p>
+									<input type="hidden" name="tribe_sie_action" value="reset_settings"/>
+									<?php wp_nonce_field( 'tribe_sie_import_nonce', 'tribe_sie_import_nonce' ); ?>
+									<?php submit_button( esc_html__( 'Reset', 'tribe-ext-settings-import-export' ), 'secondary', 'submit', false ); ?>
+								</p>
+							</form>
+						</div><!-- .inside -->
+					</div><!-- .postbox -->
+				</div><!-- .metabox-holder -->
+
+			</div><!--end .wrap-->
+
+			<?php
+		}
+
+		/**
+		 * Process a settings export that generates a .json file of the shop settings
+		 */
+		function tribe_sie_process_settings_export() {
+			// Bail if no action
+			if ( empty( $_POST['tribe_sie_action'] ) || 'export_settings' != $_POST['tribe_sie_action'] ) {
+				return;
+			}
+			// Bail if no nonce
+			if ( ! wp_verify_nonce( $_POST['tribe_sie_export_nonce'], 'tribe_sie_export_nonce' ) ) {
+				return;
+			}
+			// Bail if no capability
+			if ( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+
+			$settings = get_option( 'tribe_events_calendar_options' );
+
+			ignore_user_abort( true );
+			nocache_headers();
+			header( 'Content-Type: application/json; charset=utf-8' );
+			header( 'Content-Disposition: attachment; filename=tribe-settings-export-' . date( 'm-d-Y' ) . '.json' );
+			header( "Expires: 0" );
+			echo json_encode( $settings );
+			exit;
+		}
+
+		/**
+		 * Process a settings import from a json file
+		 */
+		function tribe_sie_process_settings_import() {
+
+			// Bail if no action
+			if ( empty( $_POST['tribe_sie_action'] ) || 'import_settings' != $_POST['tribe_sie_action'] ) {
+				return;
+			}
+
+			// Bail if no nonce
+			if ( ! wp_verify_nonce( $_POST['tribe_sie_import_nonce'], 'tribe_sie_import_nonce' ) ) {
+				return;
+			}
+
+			// Bail if no capability
+			if ( ! current_user_can( 'manage_options' ) ) {
+				return;
+			}
+
+			$import_file = $_FILES['import_file']['tmp_name'];
+			$import_filename = $_FILES['import_file']['name'];
+
+			if ( empty( $import_file ) ) {
+				wp_die( __( 'Please upload a file to import.', 'tribe-ext-settings-import-export' ) );
+			}
+
+			if ( ! empty( $import_filename ) ) {
+				$tmp = explode( '.', $import_filename );
+				$extension = end( $tmp );
+			}
+
+			if ( ! isset ( $extension ) || $extension != 'json' ) {
+				wp_die( __( 'Please upload a valid .json file.', 'tribe-ext-settings-import-export' ) );
+			}
+
+			// Retrieve the settings from the file and convert the json object to an array.
+			$settings = json_decode( file_get_contents( $import_file ), true );
+
+			if ( false === $settings ) {
+				wp_die( __( 'Sorry, we could  not decode the file.', 'tribe-ext-settings-import-export' ) );
```

Comment:
Super minor: extra space (don't _have_ to fix it):
```suggestion
				wp_die( __( 'Sorry, we could not decode the file.', 'tribe-ext-settings-import-export' ) );
```